assignment4/day 5/files/flatten.js

```diff
@@ -0,0 +1,16 @@
+function flatten(unflatObject) {
+  var output = {};
+  for (var key in unflatObject) {
+    if ((typeof unflatObject[key]) == 'object') {
+      var flatObject = flatten(unflatObject[key]);
+      for (var lastObjectKey in flatObject) {
+        output[key + '.' + lastObjectKey] = flatObject[lastObjectKey];
+      }
+    } else {
+      output[key] = unflatObject[key];
+    }
+  }
+  return output;
+}
+
+module.exports.flatten = flatten;
```

Comment:
In this case, I had used recursive approach to solve this problem and when I was using "module.exports.functionName= function(){}"  pattern, program was not working properly. that's why I had used another pattern of "module.exports" here.